app/controllers/scenarios_controller.rb

```diff
@@ -95,6 +95,20 @@ def update
     end
   end
 
+  def enable_or_disable_all_agents
+    @scenario = current_user.scenarios.find(params[:id])
+
+    respond_to do |format|
+      if @scenario.agents.update_all disabled: (params[:scenario][:disabled]) == 'true' ? true : false)
```

Comment:
Also, I don't believe that update_all can return false, so the `else` case will never get executed?